src/line2d.jl

```diff
@@ -104,3 +104,50 @@ function xiaolin_wu{T<:Gray}(img::AbstractArray{T, 2}, y0::Int, x0::Int, y1::Int
     end
     img
 end
+
+"""
+```
+img_with_line = line_normal(img, τ, color, method)
+img_with_line = line_normal(img, ρ, θ, color, method)
+```
+
+Draws a line on the input image given tuple `τ` of the form (ρ,θ) for the normal form of line:
+    x*cos(θ) + y*sin(θ) = ρ
+with the given `color`. Lines are drawn using the `bresenham` method by default. If anti-aliasing
+is required, the `xiaolin_wu` can be used.
+"""
+line_normal{T<:Colorant}(img::AbstractArray{T, 2}, args...) = line_normal!(copy(img), args...)
+
+line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, τ::Tuple{Number,Number}, method::Function = bresenham) = 
+    line_normal!(img, τ[1], τ[2], one(T), method)
+
+line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, τ::Tuple{Number,Number}, color::T, method::Function = bresenham) = 
+    line_normal!(img, τ[1], τ[2], color, method)
+
+line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, ρ::Number, θ::Number, method::Function = bresenham) =
+    line_normal!(img, ρ, θ, one(T), method)
+
+function line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, ρ::Number, θ::Number, color::T, method::Function = bresenham)
+    h,w = size(img)
```

Comment:
One of my next projects is to provide better support for visualizations with [arrays that are indexed at arbitrary values](http://docs.julialang.org/en/latest/devdocs/offset-arrays.html#Arrays-with-custom-indices-1); this is quite useful for, e.g., image transformations (rotation, warping, etc), where having arbitrary indices [is helpful for keeping track of actual location](http://juliaimages.github.io/latest/indexing.html).

Can this be changed to `indsy, indsx = indices(img)`?